mteb/tasks/Classification/eng/LinguisticProbing.py

```diff
@@ -0,0 +1,454 @@
+from __future__ import annotations
+
+from mteb.abstasks import AbsTaskClassification
+from mteb.abstasks.TaskMetadata import TaskMetadata
+
+
+class SentLenClassification(AbsTaskClassification):
```

Comment:
I think classification is fine because sentence lengths are eventually like class labels. Regression is more difficult than classification with deep neural networks.

Reply:
How is it more difficult? You just take the embeddings and train a regression head, LR could be fine but if we feel spicy could do an `MLPRegressor` with one hidden layer. I'm also thinking it would make a bit more sense, hence it's good to be able to __derive__ linguistic properties from embeddings, but good embeddings should _mostly_ encode semantics.
I would expect a kNN classifier to do waaay better on semantics, and not so good on syntax even when the model encodes it well.

Reply:
In any case, I would probably change this task to use logistic regression instead of kNN, that's probably the least painful option :)

Reply:
They provide results for logistic regression in the paper as well.

Reply:
Oh sorry I thought kNN was the default classification method on the AbsTask, don't worry about it then.